File: Pset 6/caesar.py
#!/usr/bin/env python3

import cs50
import sys

def main():
    
    # check whether we have the correct number of command-line arguments
    if len(sys.argv) != 2:
        print("Missing command-line argument, please enter the cipher key.")
        exit(1)
    
    # get cipher key from the command-line argument, convert to an integer
    key  = int(sys.argv[1]) # atoi returns 0 if given a non-integer string
    
    # prompt user for the plaintext message
    print("plaintext: ", end="")
    plaintext = cs50.get_string()
    
    # print label for the ciphertext
    print("ciphertext: ", end="")
    
    # encipher the message using Caesar's algorithm --> ci = (pi + k) % 26
    # iterate through each character in the string
    for char in plaintext:
        if char.isalpha():
            if char.isupper():
                print(chr(((ord(char) - 65 + key) % 26) + 65), end="")
            else:
                print(chr(((ord(char) - 97 + key) % 26) + 97), end="")
        else:
            print(char, end="")
        
    print()
    exit(0)
    
if __name__ == "__main__":
    main()

File: Pset 2/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    // check whether we have the correct number of command-line arguments
    if (argc != 2)
    {
        printf("Missing command-line argument, please enter the cipher key.\n");
        return 1;
    }
    
    // get cipher key from the command-line argument, convert to an integer
    int key  = atoi(argv[1]); // atoi returns 0 if given a non-integer string
    
    // prompt user for the plaintext message
    printf("plaintext: ");
    string plaintext = get_string();
    
    // print label for the ciphertext
    printf("ciphertext: ");
    
    // encipher the message using Caesar's algorithm --> ci = (pi + k) % 26
    // iterate through each character in the string
    for (int i = 0, length = strlen(plaintext); i < length; i++)
    {
        // check if the character is uppercase, then encipher
        if (isupper(plaintext[i]))
        {
            // get the i'th character's ascii value, add the key
            // subtract 65 to get the alphanumeric value
            // get the remainder after dividing by 26
            // add 65 to get the character's ascii value
            printf("%c", (((plaintext[i] + key) - 65) % 26) + 65);
        }
        
        // check if the character is lowercase, then encipher
        else if (islower(plaintext[i]))
        {
            // get the i'th character's ascii value, add the key
            // subtract 97 to get the alphanumeric value
            // get the remainder after dividing by 26
            // add 97 to get the character's ascii value
            printf("%c", (((plaintext[i] + key) - 97) % 26) + 97);
        }
        
        // if the character is not alphabetic, just print it as is
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    
    printf("\n");
    return 0;
}
